python/pyspark/sql/tests.py

```diff
@@ -2592,6 +2592,21 @@ def test_create_dataframe_from_array_of_long(self):
         df = self.spark.createDataFrame(data)
         self.assertEqual(df.first(), Row(longarray=[-9223372036854775808, 0, 9223372036854775807]))
 
+    @unittest.skipIf(not _have_pandas, "Pandas not installed")
```

Comment:
Hi, @cloud-fan and @BryanCutler .
This seems to break `branch-2.2`, but has been hidden behind another SQL error. (cc @gatorsmile , @henryr)
Please see [this](https://github.com/apache/spark/pull/19701#issuecomment-343037369).

cc @felixcheung since he is RM for 2.2.1.

Reply:
Thanks @dongjoon-hyun for tracking this down.  It looks like sql/tests.py for branch-2.2 is just missing the following

```

_have_pandas = False
try:
    import pandas
    _have_pandas = True
except:
    # No Pandas, but that's okay, we'll skip those tests
    pass
```
This was probably adding from an earlier PR in master and wasn't included when this was cherry-picked.

Reply:
I can add a patch a little bit later tonight unless someone is able to get to it first.

Reply:
I can take it over. I'll submit a pr soon.

Reply:
Thank you, @BryanCutler !

Reply:
Great, @ueshin ! :)

Reply:
BTW, @ueshin .
`branch-2.2` Jenkins will fail due to #19701 .
Could you merge #19701 first?

Reply:
Ah, in that case, maybe we need to revert one of the two original patches and fix one by one, or merge the two follow-ups into one as a hot-fix pr. cc @gatorsmile @cloud-fan